assets/model_monitoring/components/src/action_analyzer/action_analyzer_relevance_action_detector/relevance_action_detector.py

```diff
@@ -0,0 +1,606 @@
+    GROUP_TOPIC_MIN_SAMPLE_SIZE,
+    RETRIEVAL_QUERY_TYPE_COLUMN,
+    RETRIEVAL_TOP_K_COLUMN,
+    GOOD_METRICS_VALUE,
+    DEFAULT_TOPIC_NAME,
+    PROMPT_FLOW_INPUT_COLUMN,
+    DOCUMENT_RELEVANCE_SCORE_COLUMN,
+    RETRIEVAL_DOC_COLUMN,
+    MODIFIED_PROMPT_COLUMN,
+    QUERY_INTENTION_COLUMN,
+    API_CALL_RETRY_BACKOFF_FACTOR,
+    API_CALL_RETRY_MAX_COUNT,
+    ACTION_DESCRIPTION,
+    MAX_SAMPLE_SIZE,
+    DEFAULT_RETRIEVAL_SCORE,
+    P_VALUE_THRESHOLD
+)
+from shared_utilities.prompts import BERTOPIC_DEFAULT_PROMPT, RELEVANCE_TEMPLATE, RELEVANCE_METRIC_TEMPLATE
+from shared_utilities.span_tree_utils import SpanTree
+
+from shared_utilities.io_utils import (
+    np_encoder
+)
+from shared_utilities.llm_utils import (
+    API_KEY,
+    AZURE_OPENAI_API_COMPLETION_URL_PATTERN,
+    AZURE_ENDPOINT_DOMAIN_VALID_PATTERN_RE,
+    _APITokenManager,
+    _WorkspaceConnectionTokenManager,
+    _HTTPClientWithRetry,
+    _check_and_format_azure_endpoint_url,
+    _request_api,
+    get_llm_request_args
+)
+
+from bertopic import BERTopic
+from openai import AzureOpenAI
+from bertopic.representation import OpenAI
+from typing import Tuple
+
+
+class Action:
+    """Action class."""
+
+    def __init__(self, action_type, query_ids, query_intention, confidence_score):
+        """Create an action."""
+        self.action_type = action_type
+        self.query_ids = query_ids
+        self.query_intention = query_intention
+        self.confidence_score = confidence_score
+
+
+def _query_relevance_score(
+    turns: Tuple[str, str, str],
+    session: requests.Session,
+    endpoint_url: str,
+    token_manager: _APITokenManager,
+    model: str,
+    temperature: float,
+    top_p: float,
+    num_samples: int,
+    frequency_penalty: float,
+    presence_penalty: float,
+    max_tokens=3000,
+    stop: str = None
+) -> int:
+
+    turns_list = [turns]
+    prompts = [RELEVANCE_METRIC_TEMPLATE.replace("{{question}}", turn[0]).replace("{{answer}}", turn[1]).replace("{{context}}", turn[2]) for turn in turns_list]  # noqa: E501
+
+    ratings = []
+    for prompt in prompts:
+        request_data = {
+            "model": model,
+            "temperature": temperature,
+            "top_p": top_p,
+            "n": num_samples,
+            "max_tokens": max_tokens,
+            "frequency_penalty": frequency_penalty,
+            "presence_penalty": presence_penalty,
+            "messages": [
+                {"role": "user", "content": prompt}
+            ]
+        }
+
+        if stop:
+            request_data["stop"] = stop
+
+        response = {}
+        try:
+            response, time_taken = _request_api(
+                session=session,
+                endpoint_url=endpoint_url,
+                token_manager=token_manager,
+                **request_data,
+            )
+            # Append time taken to the line
+            response["response_time_sec"] = time_taken
+            rating = float(response["samples"][0])
```

Comment:
why this one does not need _proces_process_results?

and most of these lines here are the same as in the _queyr_retrieval_score, except  for prompt part. shall we refactor the code?